File: crypto/src/main/java/io/oversec/one/crypto/ui/SecureBaseActivity.java
package io.oversec.one.crypto.ui;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.WindowManager;
import io.oversec.one.common.MainPreferences;

public abstract class SecureBaseActivity extends BaseActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (!MainPreferences.isAllowScreenshots(this)) {
            getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE,
                    WindowManager.LayoutParams.FLAG_SECURE);
        }
    }
}

File: crypto/src/main/java/io/oversec/one/crypto/ui/SecureBaseActivity.kt
package io.oversec.one.crypto.ui

import android.os.Bundle
import android.view.WindowManager
import io.oversec.one.common.MainPreferences

abstract class SecureBaseActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (!MainPreferences.isAllowScreenshots(this)) {
            window.setFlags(
                WindowManager.LayoutParams.FLAG_SECURE,
                WindowManager.LayoutParams.FLAG_SECURE
            )
        }
    }
}
